### PR TITLE
Fixed mark() implementation of ResumableInputStream

### PR DESCRIPTION
**Pull request by frsyuki**
Actually removed because it is completely optional to implement.

### CODE REVIEW

File: digdag-plugin-utils/src/test/java/io/digdag/util/ResumableInputStreamTest.java [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        @Override
+        public int read() throws IOException
+        {
+            return throwIfErrorCommand(returnValues.removeFirst());
+        }
+
+        @Override
+        public int read(byte[] b, int off, int len) throws IOException
+        {
+            return throwIfErrorCommand(returnValues.removeFirst());
```
Comment (komamitsu) [inline digdag-plugin-utils/src/test/java/io/digdag/util/ResumableInputStreamTest.java]: Returning a number more than `len` can confuse me a bit. `skip()` is as well.

File: digdag-plugin-utils/src/test/java/io/digdag/util/ResumableInputStreamTest.java [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        SimpleReopener reopener = new SimpleReopener(next);
+        ResumableInputStream in = new ResumableInputStream(next, reopener);
+        assertThat(in.read(), is(1));
+        assertThat(reopener.getLastOffset(), is(0L));
+        assertThat(reopener.getReopenCount(), is(5));
+        assertThat(in.read(), is(2));
+        assertThat(reopener.getLastOffset(), is(1L));
+        assertThat(reopener.getReopenCount(), is(6));
+        assertThat(in.read(), is(-1));
+        assertThat(reopener.getReopenCount(), is(6));
```
Comment (komamitsu) [inline digdag-plugin-utils/src/test/java/io/digdag/util/ResumableInputStreamTest.java]: It seems you can check the last offset here.

**Comment by komamitsu** [comment]
Added some minor comments on ResumableInputStreamTest, but totally LGTM 👍 

BTW, `in` of ResumableInputStream can be null without synchronization at `reopen()`. So other methods (e.g. `close()`) may run into NPE in multithreads environment. Maybe we'd better synchronize those methods in ResumableInputStream later to avoid potential problems.

**Comment by frsyuki** [comment]
@komamitsu  Good points! I added a follow-up commit: https://github.com/treasure-data/digdag/commit/18d9012aa27806efafcee37f648ca78d72d4b1de

**Pull request merged**